lib/mayaHydra/hydraExtensions/CMakeLists.txt

```
@@ -30,31 +30,54 @@ set(HEADERS
 # -----------------------------------------------------------------------------
 # compiler configuration
 # -----------------------------------------------------------------------------
-target_compile_definitions(${TARGET_NAME} 
-    PRIVATE 
-        MAYAHYDRALIB_EXPORT
-        $<$<BOOL:${IS_MACOSX}>:OSMac_>
-        # Copy-pasted from lib/mayaUsd/CMakeLists.txt
-)
-
+if(DEFINED MAYAUSD_LOCATION)
+    target_compile_definitions(${TARGET_NAME} 
+        PRIVATE 
+            MAYAHYDRALIB_EXPORT
+            MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD
+            $<$<BOOL:${IS_MACOSX}>:OSMac_>
+            # Copy-pasted from lib/mayaUsd/CMakeLists.txt
+    )
+else()
+    target_compile_definitions(${TARGET_NAME} 
+        PRIVATE 
+            MAYAHYDRALIB_EXPORT
+            $<$<BOOL:${IS_MACOSX}>:OSMac_>
+            # Copy-pasted from lib/mayaUsd/CMakeLists.txt
+    )
+endif()
```

Once you create the FindMayaUsd.cmake it will set MAYAUSD_FOUND (to TRUE for cmake). Then this can be simplified with a cmake generator expression.
```suggestion
target_compile_definitions(${TARGET_NAME} 
    PRIVATE 
        MAYAHYDRALIB_EXPORT
        $<$<BOOL:${IS_MACOSX}>:OSMac_>
        $<$<BOOL:${MAYAUSD_FOUND}>:MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD>
        # Copy-pasted from lib/mayaUsd/CMakeLists.txt
)
```